### PR TITLE
EVG-16907: fix temporary nil interface check

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-16907

### Description 
I forgot about [the nil interface checking rule](https://mangatmodi.medium.com/go-check-nil-interface-the-right-way-d142776edef1), so checking for a non-nil Secrets Manager client is insufficient because an interface (`cocoa.SecretsManagerClient`) wrapping a nil pointer (`*BasicSecretsManagerClient`) is not nil. It seems like setting up container secrets _should_ happen (or else the project can't run container tasks), but we have to wait until Secrets Manager is available to production.

### Testing 
Not really testable, so I just did it by inspection.

### CODE REVIEW

File: rest/data/project.go [hunk 1]
```diff
@@ -169,6 +129,51 @@ func CreateProject(ctx context.Context, env evergreen.Environment, projectRef *m
...
+	smClient, err := cloud.MakeSecretsManagerClient(settings)
+	if err != nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    errors.Wrap(err, "setting up Secrets Manager client to store newly-created project's container secrets").Error(),
+		}
+	}
+	defer smClient.Close(ctx)
+
+	vault, err := cloud.MakeSecretsManagerVault(smClient)
```
Comment (ablack12) [inline rest/data/project.go:144]: Previously we only did this if smClient wasn't nil; do we still need to do that?

File: rest/data/project.go [hunk 2]
```diff
@@ -169,6 +129,51 @@ func CreateProject(ctx context.Context, env evergreen.Environment, projectRef *m
...
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    errors.Wrapf(err, "updating project ref's container secrets").Error(),
+		}
+	}
+	// This updates the container secrets in the DB project ref only, not
+	// the in-memory copy.
+	if err := UpsertContainerSecrets(ctx, vault, pRef.ContainerSecrets); err != nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
```
Comment (ablack12) [inline rest/data/project.go:169]: If this function is always going to return a StatuInternalServerError and we're only logging it then there's not really a reason to just return the error.Wraps directly.

File: rest/data/project.go (hunk 1 above)
Comment (Kimchelly) [reply rest/data/project.go:141]: `cloud.MakeSecretsManagerClient` will return either an error or a non-nil object, but not both, so we don't need to check for nil here.

File: rest/data/project.go (hunk 2 above)
Comment (Kimchelly) [reply rest/data/project.go]: True, I can remove all the error responses.

**Review by ablack12 — APPROVED** [review]

**Review by bynn — APPROVED** [review]

**Review by hadjri — APPROVED** [review]

**Pull request merged**